### PR TITLE
Breakdown of Vanilla gui: Improvements, added hunger widget

### PR DESCRIPTION
**Pull request by tenowg**
Signed-off-by:Craig Russell tenowg@thedemgel.com

### CODE REVIEW

**Comment by windy1** [comment]
Cannot be merged; needs a rebase.

**Comment by tenowg** [comment]
Rebase - compiled and tested

File: src/main/java/org/spout/vanilla/plugin/component/player/HUDComponent.java [hunk 1]
```diff
@@ -102,36 +104,68 @@ public void setDefault(Class<? extends GUIWidget> clazz, boolean force) {
...
 			}
 			crosshairWidget = (CrosshairWidget) widget;
 		} else if (widget instanceof DrowningWidget) {
 			if (drowningWidget != null && force == false) {
-				return;
+				return (T)drowningWidget;
 			}
 			drowningWidget = (DrowningWidget) widget;
 		}
+		else if (widget instanceof HungerWidget) {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/plugin/component/player/HUDComponent.java]: Formatting:

``` java
}
else {
```

should be 

``` java
} else {
```

**Comment by tenowg** [comment]
fixed format issue

**Comment by tenowg** [comment]
Fixed a NPE with adding non-core Widget

**Comment by Sleaker** [comment]
I just redid Quickbars, and some of the inventory API. You'll probably need to rebase and fix the references to PlayerQuickbar for this to be auto-mergable.

**Comment by windy1** [comment]
Nope @Sleaker it can be merged.

**Comment by tenowg** [comment]
Are there anymore issues related to this that I need to fix?

**Comment by windy1** [comment]
Now it needs a rebase @tenowg I promise I'll take a serious look at it once you rebase. :P

**Comment by tenowg** [comment]
Seems I need to fix something because of Quickbar

**Comment by tenowg** [comment]
rebased again ;) and fixed to reflect new QuickbarInventory - tested

**Pull request merged**